Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -461,23 +461,17 @@ public class GraphTraverser: GraphTraversing {
         }
         references.formUnion(directSystemLibrariesAndFrameworks)
 
-        // Precompiled libraries and frameworks
-        let precompiled = graph.dependencies[.target(name: name, path: path), default: []]
-            .lazy
-            .filter(\.isPrecompiled)
-
-        let precompiledDependencies =
-            precompiled
-                .flatMap { filterDependencies(from: $0) }
-
-        let precompiledDynamicLibrariesAndFrameworks = Set(precompiled + precompiledDependencies)
-            .filter(\.isPrecompiledDynamicAndLinkable)
+        let precompiledDynamicLibrariesAndFrameworks = precompiledDynamicLibrariesAndFrameworks(
+            path: path,
+            name: name
+        )
 
         let staticXCFrameworksLinkedByDynamicXCFrameworkDependencies = filterDependencies(
-            from: Set(precompiledDynamicLibrariesAndFrameworks).filter {
-                $0.xcframeworkDependency != nil
+            from: Set(precompiledDynamicLibrariesAndFrameworks).filter { $0.xcframeworkDependency != nil },
+            test: {
+                $0.xcframeworkDependency?.linking == .static &&
+                    $0.xcframeworkDependency?.path.glob("**/*.swiftmodule").isEmpty == false
```

Comment:
I'd be consistent with other metadata associated with an `.xcframework` and load this information in memory when the XCFramework dependency is loaded (e.g. `containsSwiftModules`, `containsModuleMaps`)

Reply:
I'm not sure I agree. Why load this information _always_ in memory if we need it in a single place? And increasing the API surface of `XcodeGraph` for something I'd consider very niche?

Reply:
> I'm not sure I agree. Why load this information always in memory if we need it in a single place?

A quick search of `**/*.swiftmodule` and `**/*.modulemap` reveals two instances of the logic in the PR. 

Most of the metadata we load for XCFrameworks could be considered a niche since we'll be the main consumers of `XcodeGraph`, even after we extract the generation logic. My concern is that if we don't draw a line, we'll have a scattering of properties across the two repositories, so future contributions will find themselves having to decide where something should go.

Reply:
> My concern is that if we don't draw a line, we'll have a scattering of properties across the two repositories, so future contributions will find themselves having to decide where something should go.

Agree, but that doesn't mean shoving everything into `XcodeGraph`. We should limit the amount of PRs we need to do for `XcodeGraph` as doing cross-repo PRs is painful. This super tight coupling is not great and we should aim for `XcodeGraph` to be more generic and thus stable. If we're to put every little change into `XcodeGraph` for a niche use case of `tuist`, it's gonna be very hard to keep `XcodeGraph` stable.

Reply:
> We should limit the amount of PRs we need to do for XcodeGraph as doing cross-repo PRs is painful

This is a bad sign to me. If we refrain from adding changes to the model because of the pain of our architecture, we should either make that less painful or eliminate the pain. 

When I look at the [metadata](https://github.com/tuist/XcodeGraph/blob/12462c8fff97fc5efb8476ef4625f434acd5058f/Sources/XcodeGraph/Models/Metadata/XCFrameworkMetadata.swift#L6), it's not obvious to me which ones of those properties should be there because they are not used in a "niche" use case, and which ones should not because they are. 

Since the direction of XcodeGraph and the idea of extracting the generation logic is not clear, let's discuss that separately, merge this PR, and iterate on this if needed based on the conclusion of that discussion.